src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/LayersNamesAliasingTests.java

```diff
@@ -0,0 +1,229 @@
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.w3c.dom.Document;
+
+/**
+ * Tests that validate that layer names aliasing works correctly for App-Schema defined feature
+ * types, i.e. that is possible to create a layer with a name different from the App-Schema feature
+ * type.
+ */
+public final class LayersNamesAliasingTests extends GeoServerSystemTestSupport {
+
+    // xpath engines used to check WFS responses
+    private XpathEngine WFS11_XPATH_ENGINE;
+    private XpathEngine WFS20_XPATH_ENGINE;
+
+    private static Path testFolderPath;
+    private static Path mappingsFolderPath;
+
+    @BeforeClass
+    public static void prepare() throws IOException {
+        testFolderPath = Files.createTempDirectory(Paths.get("target/test-classes"), "layernames");
+        File srcDir = new File("target/test-classes/test-data/stations/layerNamesTest");
+        File destDir = Paths.get(testFolderPath.toString(), "layerNamesTest").toFile();
+        FileUtils.copyDirectory(srcDir, destDir);
+        mappingsFolderPath = destDir.toPath();
+    }
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+        super.onSetUp(testData);
+        Catalog catalog = getCatalog();
+        WorkspaceInfoImpl workspace = new WorkspaceInfoImpl();
+        workspace.setName("st");
+        NamespaceInfoImpl nameSpace = new NamespaceInfoImpl();
+        nameSpace.setPrefix("st");
+        nameSpace.setURI("http://www.stations.org/1.0");
+        catalog.add(workspace);
+        catalog.add(nameSpace);
+        // create the app-schema data store
+        Map<String, Serializable> params = new HashMap<>();
+        params.put("dbtype", "app-schema");
+        params.put(
+                "url",
+                Paths.get(mappingsFolderPath.toString(), "stationsDefaultGeometry.xml")
+                        .toUri()
+                        .toString());
+        DataStoreInfoImpl dataStore = new DataStoreInfoImpl(getCatalog());
+        dataStore.setId("stations");
+        dataStore.setName("stations");
+        dataStore.setType("app-schema");
+        dataStore.setConnectionParameters(params);
+        dataStore.setWorkspace(workspace);
+        dataStore.setEnabled(true);
+        catalog.add(dataStore);
+        // build the feature type for the root mapping (StationFeature)
+        CatalogBuilder builder = new CatalogBuilder(catalog);
+        builder.setStore(dataStore);
+        builder.setWorkspace(workspace);
+        FeatureTypeInfo featureType =
+                builder.buildFeatureType(new NameImpl(nameSpace.getURI(), "Station"));
+        featureType.setName("MyStation");
+        featureType.setNativeName("Station");
+        catalog.add(featureType);
+        LayerInfo layer = builder.buildLayer(featureType);
+        layer.setDefaultStyle(catalog.getStyleByName("point"));
+        catalog.add(layer);
+    }
+
+    @Test
+    public void testMyStationGet() throws Exception {
```

Comment:
I would suggest as test name ``testAliasedNameWfsGetFeature11`` instead of ``testMyStationGet``.